csharp/src/Ice/IObject.cs

```diff
@@ -115,4 +116,42 @@ protected ValueTask<OutgoingResponseFrame> IceD_ice_idsAsync(IncomingRequestFram
                                                           ostr.WriteSequence(ret, OutputStream.IceWriterFromString)));
         }
     }
+
+    internal static class Servant
+    {
+        internal static ValueTask<OutgoingResponseFrame> DispatchWithInterceptorsAsync(
```

Comment:
I don't like this dynamic and recursive building of a new interceptor chain for every single dispatch. I think interceptors / interceptor chain should be much more static - set and frozen when the object adapter is activated.

Then, does it make sense to have:
- a public servant.DispatchAsync which bypasses interceptors
- an internal servant.DispatchWithInterceptorAsync that goes through the interceptor chain

It seems to me the default should be to use the interceptors, so if we have 2 APIs with one internal and the other one public, it should be the other way around.